library/core/src/marker.rs

```diff
@@ -482,6 +482,28 @@ impl<T: ?Sized> !Sync for *const T {}
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: ?Sized> !Sync for *mut T {}
 
+#[unstable(feature = "gc", issue = "none")]
+#[cfg_attr(not(test), rustc_diagnostic_item = "FinalizerSafe")]
+/// Types for which it is safe to use inside a finalizer.
```

Comment:
Maybe this first sentence should be something like  "A marker trait used to state that a struct's drop method can safely be used with GC finalisation." ?